### PR TITLE
fix: correct the logic of delete model

### PR DESCRIPTION
**Pull request by frostming**
Signed-off-by: Frost Ming <me@frostming.com>

## What does this PR address?

Previously, when the model version is missing, model_store.delete() will delete the **whole** `$model_name` directory. And if the version is `latest`, it will raise an error since `$model_name/latest` isn't a directory.

Fix this to resolve to the actual version to get the path correctly.

## Before submitting:

- [ ] Does the Pull Request follow [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary) naming? Here are [GitHub's
      guide](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) on how to create a pull request.
- [ ] Does the code follow BentoML's code style, `pre-commit run -a` script has passed ([instructions](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#style-check-auto-formatting-type-checking))?
- [ ] Did you read through [contribution guidelines](https://github.com/bentoml/BentoML/blob/main/CONTRIBUTING.md#ways-to-contribute) and follow [development guidelines](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#start-developing)?
- [ ] Did your changes require updates to the documentation? Have you updated
      those accordingly? Here are [documentation guidelines](https://github.com/bentoml/BentoML/tree/main/docs) and [tips on writting docs](https://github.com/bentoml/BentoML/tree/main/docs#writing-documentation).
- [ ] Did you write tests to cover your changes?

### CODE REVIEW

File: src/bentoml_cli/models.py [hunk 1]
```diff
@@ -254,7 +254,6 @@ def import_from(model_path: str) -> None:  # type: ignore (not accessed)
         "--context", type=click.STRING, default=None, help="Yatai context name."
     )
     @click.option(
-        "-f",
```
Comment (frostming) [inline src/bentoml_cli/models.py:257]: Conflicting short option name with `--force`

Comment (sauyon) [reply src/bentoml_cli/models.py:257]: Let's just rename this to `-F`?

File: src/bentoml/_internal/store.py [hunk 2]
```diff
@@ -182,6 +182,13 @@ def register(self, tag: t.Union[str, Tag]):
...
         _tag = Tag.from_taglike(tag)
 
+        if _tag.version in (None, "latest"):
+            try:
+                _tag.version = self._fs.readtext(_tag.latest_path())
+            except fs.errors.ResourceNotFound:
+                # if latest path doesn't exist, we don't need to delete anything
+                return
+
         if not self._fs.exists(_tag.path()):
```
[the remark this answers is not in the document]
Comment (frostming) [reply src/bentoml/_internal/store.py:192]: I thought it a bit further, we don't need to delete the latest version, but allow to update the latest version from the remote store.

[the remark this answers is not in the document]
Comment (aarnphm) [reply src/bentoml/_internal/store.py]: I see. that makes sense as well.

File: src/bentoml/_internal/cloud/yatai.py [hunk 3]
```diff
@@ -873,19 +873,21 @@ def _do_pull_model(
...
+            model = None
+        else:
+            if _tag.version not in (None, "latest"):
+                if not force:
+                    self.log_progress.add_task(
+                        f'[bold blue]Model "{tag}" already exists locally, skipping'
+                    )
+                    return model
+                else:
+                    model_store.delete(tag)
```
Comment (aarnphm) [inline src/bentoml/_internal/cloud/yatai.py:889]: You will need to also add this to `_internal/cloud/bentocloud.py` as well 😄

Comment (frostming) [reply src/bentoml/_internal/cloud/yatai.py:889]: Oh yes, may be we could make it DRYer in the future.

Comment (aarnphm) [reply src/bentoml/_internal/cloud/yatai.py:889]: I think the idea for yatai.py and bentocloud.py is to have different implementation, but currently the bentocloud push for both model and bento are the same with yatai

**Review by aarnphm — APPROVED** [review]

**Comment by aarnphm** [comment]
I Think we should have an option with `bentoml delete --all` or `bentoml prune` to just remove all the bentos and models available locally

**Pull request merged**